packages/clerk-js/src/ui/elements/PasswordInput.tsx

```diff
@@ -1,34 +1,110 @@
 import type { ChangeEvent } from 'react';
+import { useCallback, useRef } from 'react';
 import React, { forwardRef } from 'react';
 
 import { useEnvironment } from '../contexts';
 import { descriptors, Flex, Input } from '../customizables';
-import { usePasswordComplexity } from '../hooks';
+import { usePasswordComplexity, usePasswordStrength } from '../hooks';
 import { EyeSlash } from '../icons';
 import { useFormControl } from '../primitives/hooks';
 import type { PropsOfComponent } from '../styledSystem';
 import { IconButton } from './IconButton';
 
 type PasswordInputProps = PropsOfComponent<typeof Input> & {
+  strengthMeter?: boolean;
   complexity?: boolean;
 };
 
+/**
+ * Coordinate the success and error states between password complexity and strength
+ * `usePasswordStrength` and `usePasswordComplexity` will fire callbacks to handle error state for their own
+ * cases.
+ * This hook aims to synchronize those states and throw 1 error if such exists
+ * - Complexity errors has priority over strength errors.
+ * - To set a field as successful, both complexity and strength need to be successful
+ */
+const useComplexityOverStrength = ({ strengthMeter }: Required<Pick<PasswordInputProps, 'strengthMeter'>>) => {
+  const formControlProps = useFormControl();
+  const hasComplexityError = useRef(false);
+  const hasComplexitySuccess = useRef(false);
+  return {
+    setStrengthSuccess: useCallback(() => {
+      if (!hasComplexityError.current) {
+        formControlProps.setSuccessful?.(true);
+      }
+    }, []),
+
+    setStrengthError: useCallback((errorMessage: string) => {
+      if (!hasComplexityError.current) {
+        formControlProps.setError?.(errorMessage);
+      }
+    }, []),
+
+    setComplexityError: useCallback((errorMessage: string) => {
+      hasComplexityError.current = !!errorMessage;
+      formControlProps.setError?.(errorMessage);
+    }, []),
+
+    setComplexitySuccess: useCallback(() => {
+      hasComplexityError.current = false;
+      hasComplexitySuccess.current = true;
+      if (!strengthMeter) {
+        formControlProps.setSuccessful?.(true);
+      }
+    }, []),
+  };
+};
+
+const loadZxcvbn = () => {
+  return Promise.all([
+    // @ts-ignore
+    import(/*webpackIgnore: true*/ 'https://cdn.jsdelivr.net/npm/@zxcvbn-ts/core@2.2.1/+esm'),
+    // @ts-ignore
+    import(/*webpackIgnore: true*/ 'https://cdn.jsdelivr.net/npm/@zxcvbn-ts/language-common@2.0.1/+esm'),
+  ]).then(([core, zxcvbnCommonPackage]) => {
```

Comment:
What do we do if loading fails for some reason? Shall we retry or let the FAPI handle validation?

Reply:
Hmm, FAPI will not accept any password that does not meet the criteria (once submitted), having said that I don't think we want to make requests to FAPI (on keystroke).

Reply:
![Screenshot 2023-04-12 at 11 02 59 AM](https://user-images.githubusercontent.com/19269911/231393080-cf32e2af-4e1b-4d56-a5f4-95cf894d7a56.png)

In any case the FE allows the request to go through with a weak password, FAPI will catch it. But I see your point.

if(library is downloaded) {dontAllowForWeakPasswordsOnSubmit}
if(library has failed download) {allowForAnyPassword > 8 length to be submitted}